app/views/locations/edit_name.html.erb

```diff
@@ -3,7 +3,7 @@
 <% content_for :before_content do %>
   <%= govuk_back_link(
     text: "Back",
-    href: "/schemes/#{@scheme.id}/locations",
+    href: "/schemes/#{@scheme.id}/locations/#{@location.id}",
```

Comment:
```suggestion
    href: location_path(location),
```

Reply:
It routes to `/schemes/#{@location.id}/locations/#{@location.id}` if I call it like that

Reply:
Ah, the routing likely isn't right. The `resources :locations` needn't be wrapped in `member { }`.

Then you can change the routes to:

```suggestion
    href: scheme_locations_path(@scheme, @location),
```

and so on!

Reply:
Before:

```
                                                                   location_edit_name GET    /schemes/:id/locations/:location_id/edit-name(.:format)                                                            locations#edit_name
                                                                        location_edit GET    /schemes/:id/locations/:location_id/edit(.:format)                                                                 locations#edit
                                                        location_edit_local_authority GET    /schemes/:id/locations/:location_id/edit-local-authority(.:format)                                                 locations#edit_local_authority
                                                                            locations GET    /schemes/:id/locations(.:format)                                                                                   locations#index
                                                                                      POST   /schemes/:id/locations(.:format)                                                                                   locations#create
                                                                         new_location GET    /schemes/:id/locations/new(.:format)                                                                               locations#new
                                                                        edit_location GET    /schemes/:id/locations/:id/edit(.:format)                                                                          locations#edit
                                                                             location GET    /schemes/:id/locations/:id(.:format)                                                                               locations#show
                                                                                      PATCH  /schemes/:id/locations/:id(.:format)                                                                               locations#update
                                                                                      PUT    /schemes/:id/locations/:id(.:format)                                                                               locations#update
                                                                                      DELETE /schemes/:id/locations/:id(.:format)                                                                               locations#destroy
```

After:

```
                                                            scheme_location_edit_name GET    /schemes/:scheme_id/locations/:location_id/edit-name(.:format)                                                     locations#edit_name
                                                                 scheme_location_edit GET    /schemes/:scheme_id/locations/:location_id/edit(.:format)                                                          locations#edit
                                                 scheme_location_edit_local_authority GET    /schemes/:scheme_id/locations/:location_id/edit-local-authority(.:format)                                          locations#edit_local_authority
                                                                     scheme_locations GET    /schemes/:scheme_id/locations(.:format)                                                                            locations#index
                                                                                      POST   /schemes/:scheme_id/locations(.:format)                                                                            locations#create
                                                                  new_scheme_location GET    /schemes/:scheme_id/locations/new(.:format)                                                                        locations#new
                                                                 edit_scheme_location GET    /schemes/:scheme_id/locations/:id/edit(.:format)                                                                   locations#edit
                                                                      scheme_location GET    /schemes/:scheme_id/locations/:id(.:format)                                                                        locations#show
                                                                                      PATCH  /schemes/:scheme_id/locations/:id(.:format)                                                                        locations#update
                                                                                      PUT    /schemes/:scheme_id/locations/:id(.:format)                                                                        locations#update
                                                                                      DELETE /schemes/:scheme_id/locations/:id(.:format)                                                                        locations#destroy
```

Reply:
Nice one, was just looking at the same thing. Might open a separate PR for updating the routes and merge it in first. Tyty!